tests/chainerx_tests/unit_tests/routines_tests/test_indexing.py

```diff
@@ -201,3 +203,64 @@ def forward_xp(self, inputs, xp):
         else:
             b = a.take(indices, axis)
         return b,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize_pytest('cond_shape,x_shape,y_shape', [
+    # Same Shapes
+    ((2, 3), (2, 3), (2, 3)),
+    # Broadcast Shapes
+    ((2, 3), (1, 3), (1, 3)),
+    ((2, 3), (2, 1), (1, 3)),
+    ((2, 3), (2, 3), (1, 3)),
+    ((4, 5), (3, 4, 1), (1, 5)),
+    ((1, 4, 5), (3, 4, 1), (3, 1, 5)),
+])
+@chainer.testing.parameterize_pytest(
+    'in_dtypes,out_dtype', dtype_utils.result_dtypes_two_arrays
+)
+@chainer.testing.parameterize_pytest(
+    'condition_dtype', chainerx.testing.all_dtypes)
+class TestWhere(op_utils.NumpyOpTest):
+
+    check_numpy_strides_compliance = False
+
+    def setup(self):
+        (x_dtype, y_dtype) = self.in_dtypes
+        if numpy.dtype(x_dtype).kind != 'f' or \
+           numpy.dtype(y_dtype).kind != 'f':
+            self.skip_backward_test = True
+            self.skip_double_backward_test = True
+
+        if x_dtype == 'float16' or y_dtype == 'float16':
+            self.check_backward_options.update({'rtol': 1e-3, 'atol': 1e-3})
+
+    def generate_inputs(self):
+        (x_dtype, y_dtype) = self.in_dtypes
+        x = array_utils.uniform(self.x_shape, x_dtype)
+        y = array_utils.uniform(self.y_shape, y_dtype)
+        condition = array_utils.uniform(self.cond_shape, self.condition_dtype)
```

Comment:
`self.condition` should be of `self.condition_dtype`, but this `condition` should be float array.

Reply:
On it.

Reply:
Could you fix to `numpy.random.uniform(0, 1, size=self.cond_shape, 'float32')`?
`array_utils.uniform` does not guarantee the range of return value.

Reply:
Done.